lms/services/grouping.py

```diff
@@ -26,74 +26,48 @@ def upsert(self, grouping):
 
         return db_grouping or grouping
 
-    def upsert_canvas_section(
-        self, tool_consumer_instance_guid, context_id, section_id, section_name
+    @staticmethod
+    def generate_authority_provided_id(
+        tool_consumer_instance_guid, lms_id, parent: Grouping, type_: Grouping.Type
     ):
-        """
-        Upsert a Grouping for a course section.
-
-        :param tool_consumer_instance_guid: Tool consumer GUID
-        :param context_id: Course id the section is a part of
-        :param section_id: A section id for a section group
-        :param section_name: The name of the section
-        """
-
-        section_authority_provided_id = hashed_id(
-            tool_consumer_instance_guid, context_id, section_id
-        )
-
-        course = self._course_service.get(
-            self._course_service.generate_authority_provided_id(
-                tool_consumer_instance_guid, context_id
+        if type_ == Grouping.Type.CANVAS_SECTION:
+            return hashed_id(tool_consumer_instance_guid, parent.lms_id, lms_id)
```

Comment:
Sections and courses generated the authority IDs before groupings were a thing so they use the specific order of params (and don't include the type)